server/src/main/java/org/elasticsearch/health/HealthIndicator.java

```diff
@@ -8,13 +8,12 @@
 
 package org.elasticsearch.health;
 
-import org.elasticsearch.xcontent.ToXContentFragment;
 import org.elasticsearch.xcontent.ToXContentObject;
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
 
-public record HealthIndicatorResult(String name, String component, HealthStatus status, String summary, ToXContentFragment details)
+public record HealthIndicator(String name, String component, HealthStatus status, String summary, ToXContentObject details)
```

Comment:
I'd prefer to keep the name here, but happy to discuss this more closely. It matches what we did in autoscaling, where each decider has a Service and delivers a Result.